src/Language/Haskell/Exts/Bracket.hs

```diff
@@ -42,20 +42,40 @@ instance (Data l, Default l) => Brackets (Exp l) where
     addParen = Paren def
 
     isAtom x = case x of
-        Paren{}           -> True
-        Tuple{}           -> True
-        List{}            -> True
-        LeftSection{}     -> True
-        RightSection{}    -> True
-        TupleSection{}    -> True
-        RecConstr{}       -> True
-        ListComp{}        -> True
-        EnumFrom{}        -> True
-        EnumFromTo{}      -> True
-        EnumFromThen{}    -> True
-        EnumFromThenTo{}  -> True
-        OverloadedLabel{} -> True
-        _                 -> isLexeme x
+        Var{}                -> True
+        Con{}                -> True
+        Lit{}                -> True
+        Paren{}              -> True
+        Tuple{}              -> True
+        List{}               -> True
+        LeftSection{}        -> True
+        RightSection{}       -> True
+        TupleSection{}       -> True
+        RecConstr{}          -> True
+        ListComp{}           -> True
+        EnumFrom{}           -> True
+        EnumFromTo{}         -> True
+        EnumFromThen{}       -> True
+        EnumFromThenTo{}     -> True
+        OverloadedLabel{}    -> True
+        ParArray{}           -> True
+        ParComp{}            -> True
+        XTag{}               -> True
+        IPVar{}              -> True
+        UnboxedSum{}         -> True
```

Comment:
This requires a recent version of haskell-src-exts.

Reply:
Would you like CPP here?

Reply:
No, I think it's just a matter of updating the lower bounds in the cabal file, so that the build / package tooling can handle it.